### PR TITLE
Fix exception handling

### PR DESCRIPTION
**Pull request by rajat08**

### CODE REVIEW

**Comment by mutayroei** [comment]
[ch10321]

File: pinecone/core/grpc/index_grpc.py [hunk 1]
```diff
@@ -67,7 +67,10 @@ class GRPCIndexBase(ABC):
     Base class for grpc-based interaction with Pinecone indexes
     """
 
-    def __init__(self, name: str, channel=None, grpc_config: GRPCClientConfig = None, _endpoint_override: str = None):
+    _pool = None
+
+    def __init__(self, name: str, channel=None, grpc_config: GRPCClientConfig = None, _endpoint_override: str = None,
```
Comment (mutayroei) [inline pinecone/core/grpc/index_grpc.py:72]: @rajat08  I believe that for the default index the parameter is called index_name

**Comment by mutayroei** [comment]
@benjaminran  can you review?

**Comment by rajat08** [comment]
@mutayroei @benjaminran 
Since all the info form the grpc error is condensed in the debug string, we can pass this to PineconeException and get a 'nicer sounding' error. However _InactiveRPCError would be visible in the trace. Would the ideal behavior be to raise PineconeError at an earlier step?

**Review by benjaminran — CHANGES_REQUESTED** [review]
Posted a couple comments about the async logic.

The error handling logic looks good! I think it's a good thing to include the _InactiveRpcError deeper inside the trace.

**Review by mutayroei — APPROVED** [review]

**Pull request merged**